test/activerecord_spanner_mock_server/models/string_io.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class StringIO
+  def ==(o)
+    o.class == self.class && self.to_base64 == o.to_base64
+  end
+
+  def eql?(o)
+    self == o
+  end
+
+  def hash
+    to_base64.hash
+  end
+
+  def to_base64
+    self.rewind
+    value = self.read
+    Base64.strict_encode64 value.force_encoding("ASCII-8BIT")
```

Comment:
Might be a dumb question: why is `ASCII-8BIT` but not `UTF-8`?

Reply:
That is what the library was already using: https://github.com/googleapis/ruby-spanner-activerecord/blob/main/lib/active_record/type/spanner/bytes.rb

So to keep this change as small / non-intrusive as possible, I kept that.